app/models/page/DashboardPage.scala

```diff
@@ -16,25 +16,76 @@
 
 package models.page
 
-import models.ApplicationData.ApplicationStatus
-import models.ApplicationData.ApplicationStatus.ApplicationStatus
+import connectors.{AllocationExchangeObjects, ExchangeObjects}
 import models.page.DashboardPage.ProgressStepVisibility
-import models.{ ApplicationData, CachedData, Progress }
+import models._
+import org.joda.time.{DateTime, LocalDate}
+import org.joda.time.format.DateTimeFormat
 import play.api.i18n.Lang
 import play.api.mvc.RequestHeader
 import security.RoleUtils
-import security.Roles.DisplayOnlineTestSectionRole
+import security.Roles._
 
 // format: OFF
 case class DashboardPage(firstStepVisibility: ProgressStepVisibility,
                          secondStepVisibility: ProgressStepVisibility,
                          thirdStepVisibility: ProgressStepVisibility,
-                         fourthStepVisibility: ProgressStepVisibility)
+                         fourthStepVisibility: ProgressStepVisibility,
+                         isApplicationSubmittedAndNotWithdrawn: Boolean,
+                         isApplicationInProgressAndNotWithdrawn: Boolean,
+                         isApplicationWithdrawn: Boolean,
+                         isApplicationCreatedOrInProgress: Boolean,
+                         isUserWithNoApplication: Boolean,
+                         isFirstStepVisible: String,
+                         isSecondStepVisible: String,
+                         isThirdStepVisible: String,
+                         isFourthStepVisible: String,
+                         fullName: String,
+                         assessmentInProgressStatus: AssessmentStatus,
+                         assessmentCompletedStatus: AssessmentStatus
+                        ) {
+  import DashboardPage._
 
+  def toddMMMMyyyyFormat(date: LocalDate): String = ddMMMMyyyy.print(date)
```

Comment:
@henricook: Done. It's now in an helper class and unit tested (though it doesn't do much)